evaluation/dataset.py

```
@@ -0,0 +1,100 @@
+                case = TestCase.model_validate(data)
+            except (json.JSONDecodeError, Exception) as exc:
+                errors.append(f"Line {line_num}: {exc}")
+                continue
+
+            if case.question in seen_questions:
+                logger.warning("Duplicate question on line %d: %s", line_num, case.question[:60])
+            seen_questions.add(case.question)
+            cases.append(case)
+
+    if errors:
+        raise ValueError(f"Test set validation errors:\n" + "\n".join(errors))
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)